head/src/main/kotlin/io/qalipsis/core/head/web/ReportController.kt

```diff
@@ -274,4 +276,82 @@ internal class ReportController(
             )
         )
     }
+
+    @Post("/{reportReference}/render")
+    @Operation(
+        summary = "Render a report",
+        responses = [
+            ApiResponse(responseCode = "200", description = "Details of the report to be rendered"),
+            ApiResponse(responseCode = "400", description = "Invalid request supplied"),
+            ApiResponse(responseCode = "401", description = "Operation not allowed"),
+            ApiResponse(responseCode = "404", description = "report not found"),
+        ],
+        security = [
+            SecurityRequirement(name = "JWT")
+        ]
+    )
+    @Secured(value = [Permissions.READ_REPORT])
+    suspend fun render(
+        @Parameter(
+            name = "X-Tenant",
+            description = "Contextual tenant",
+            required = true,
+            `in` = ParameterIn.HEADER
+        ) @NotBlank @Tenant tenant: String,
+        @Parameter(hidden = true) authentication: Authentication,
+        @Parameter(
+            description = "Reference of the report to render",
+            required = true,
+            `in` = ParameterIn.PATH
+        ) @NotBlank @PathVariable reportReference: String
+    ): HttpResponse<String> {
+        return HttpResponse.ok(
+            reportService.render(
+                tenant,
+                authentication.name,
+                reportReference
+            )
+        )
+    }
+
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    @Get("/file/{taskReference}")
+    @Operation(
+        summary = "Downloads a report",
+        responses = [
+            ApiResponse(responseCode = "200", description = "Details of the report to be rendered"),
+            ApiResponse(responseCode = "400", description = "Invalid request supplied"),
+            ApiResponse(responseCode = "401", description = "Operation not allowed"),
+            ApiResponse(responseCode = "404", description = "report not found"),
+        ],
+        security = [
+            SecurityRequirement(name = "JWT")
+        ]
+    )
+    @Secured(value = [Permissions.READ_REPORT])
+    suspend fun download(
+        @Parameter(
+            name = "X-Tenant",
+            description = "Contextual tenant",
+            required = true,
+            `in` = ParameterIn.HEADER
+        ) @NotBlank @Tenant tenant: String,
+        @Parameter(hidden = true) authentication: Authentication,
+        @Parameter(
+            description = "Reference of the task to be downloaded",
+            required = true,
+            `in` = ParameterIn.PATH
+        ) @NotBlank @PathVariable taskReference: String
+    ): HttpResponse<ByteArray> {
+        val documentName = "Report.pdf"
```

Comment:
Please use a constant. It would be good to have the name of the ReportEntity and the date of creation (from the rendering task) in the file name.